openpype/modules/extra_addons/process_monitor/addon.py

```diff
@@ -0,0 +1,704 @@
+        There is no way to prevent addon to do invalid operations if he's
+        not handling them.
+        """
+        # Make sure dialog is created
+        self._create_dialog()
+        self._process_widget.open()
+
+    def request_start_timer(self, pid):
+        #self.log.warning("request_start_timer: {}".format(pid))
+
+        if pid is None:
+            self._timers_manager_module.close_message()
+            self.stop_timers()
+            return
+
+        #TODO: Can happen? Do what in this case?
+        if not pid in self._running_processes:
+            self.log.error((
+                "Process ID '{}' not found in running processes!"
+            ).format(pid))
+            return
+
+        #self.log.warning("CURRENT PID: {}".format(self._current_pid))
+        if self._current_pid != pid:
+            if self._timers_manager_module.is_running:
+                self.timer_stopped()
+
+                #TODO: needed?
+                import time
+                time.sleep(1)
+
+            '''
+            self._current_pid = pid
+            self.timer_started(self._running_processes[pid]["data"])
+            '''
+            self.start_process_timer(pid)
+
+        else:
+            if not self._timers_manager_module.is_running:
+                self.start_process_timer(pid)
+
+    def on_action_trigger(self):
+        """Implementation of abstract method for `ITrayAction`."""
+        self.show_dialog()
+
+    def get_plugin_paths(self):
+        """Implementation of abstract method for `IPluginPaths`."""
+        current_dir = os.path.dirname(os.path.abspath(__file__))
+
+        return {
+            "publish": [os.path.join(current_dir, "plugins", "publish")]
+        }
+
+    def cli(self, click_group):
+        click_group.add_command(cli_main)
+
+    def start_timer(self, data):
+        #self.log.info("addon - start_timer - data: {}".format(data))
+
+        project_name = data["project_name"]
+        #TODO: Avoid repeating process is same project as last time
+        # => Keep track of project name, and don't call if same project
+        self._project_applications = self._get_applications(project_name)
+        self.log.info((
+            "Registered Applications for '{}': {}"
+        ).format(project_name, self._project_applications))
+
+        self._add_running_process(data)
+
+    def stop_timer(self):
+        #self.log.info("stop_timer")
+
+        #self._current_pid = None
+
+        self._process_widget.signal_update_table.emit(self._running_processes,
+                                                      None)
+
+        # check if other running processes
+
+        #self.log.warning("stop_timer - running processes '{}'".format(self._running_processes))
+        #TODO: kill waiting threads? (NOT HERE!)
+        #self.stop_waiting_thread(pid)
+
+    def _get_applications(self, project_name):
+        project_anatomy_settings = get_anatomy_settings(project_name)
```

Comment:
You should use `ApplicationsManager` here. It's not job of this addon to know how to get applications.

BTW: `Applications` will be separated addon in future so it will be optional.